### DISCUSSION TITLE
How to get all children pages?

### DISCUSSION THREAD

**Discussion by ghost** [ramnes/notion-sdk-py#190] [Q&A] [answered]
I have the ID of a top page, and I can get data from it via `notion.pages.retrieve("…")`. However, I am struggling to find a way to get all the sub-pages from this page.

Could anyone point me in the right direction?

Thank you.

**Comment by ramnes** [comment, marked as answer]
If your page is a database, you probably want to use [`notion.databases.query()`](https://ramnes.github.io/notion-sdk-py/reference/api_endpoints/#notion_client.api_endpoints.DatabasesEndpoint.query), otherwise [`notion.blocks.children.list()`](https://ramnes.github.io/notion-sdk-py/reference/api_endpoints/#notion_client.api_endpoints.BlocksChildrenEndpoint.list).

**Reply by ghost** [reply]
I should have said, it is not a database. Just a top page. Thank you, I shall try tomorrow and report.

**Reply by ghost** [reply]
It worked fine.

Here is a simple piece of code that does what I wanted. In case that's helpful as an example for anyone else.

```python

    def _get_page_data(self, _id: str) -> None:
        page = self._notion.pages.retrieve(_id)
        title = ""
        try:
            title = page["properties"]["title"]["title"][0]["plain_text"]
        except KeyError as e:
            rlog.exception(e)
        rlog.info(
            "A page",
            uuid=page.get("id"),
            title=title,
            create_by=self._users[page.get("created_by")["id"]],
            last_edited=self._users[page.get("last_edited_by")["id"]],
        )
        blocks = self._notion.blocks.children.list(_id)["results"]
        for block in blocks:
            if block["object"] == "block" and block["type"] == "child_page":
                self._get_page_data(block["id"])
```
